icgcget/cli.py

```diff
@@ -79,123 +95,211 @@ def cli(ctx, config, logfile):
+@click.option('--gdc-access', type=click.STRING)
+@click.option('--gdc-path', type=click.Path(exists=True, dir_okay=False, resolve_path=True))
+@click.option('--gdc-transport-parallel', type=click.STRING)
+@click.option('--gdc-udt', type=click.BOOL)
+@click.option('--icgc-access', type=click.STRING)
+@click.option('--icgc-path', type=click.Path(exists=True, dir_okay=False, resolve_path=True))
+@click.option('--icgc-transport-file-from', type=click.STRING)
+@click.option('--icgc-transport-parallel', type=click.STRING)
+@click.option('--yes-to-all', '-y', is_flag=True, default=False, help="Bypass all confirmation prompts")
+@click.pass_context
+def download(ctx, repos, fileids, manifest, output,
              cghub_access, cghub_path, cghub_transport_parallel,
-             ega_access, ega_password, ega_path, ega_transport_parallel, ega_udt, ega_username,
+             ega_password, ega_path, ega_transport_parallel, ega_udt, ega_username,
              gdc_access, gdc_path, gdc_transport_parallel, gdc_udt,
-             icgc_access, icgc_path, icgc_transport_file_from, icgc_transport_memory, icgc_transport_parallel):
-    logger = logging.getLogger('__log__')
-    code = 0
+             icgc_access, icgc_path, icgc_transport_file_from, icgc_transport_parallel, yes_to_all):
+    if os.getenv("ICGCGET_API_URL"):
+        api_url = os.getenv("ICGCGET_API_URL")
+    else:
+        api_url = ctx.default_map["portal_url"] + ctx.default_map["portal_api"]
+    object_ids = {}
 
-    if fileid is None and manifest is None:
-        logger.error("Please provide either a file id value or a manifest file to download.")
-        code = 1
-        return code
+    size = 0
+    if manifest:
+        if len(fileids) > 1:
+            logger.warning("For download from manifest files, multiple manifest id arguments is not supported")
+            raise click.BadArgumentUsage("Multiple manifest files specified.")
+        try:
+            manifest_json = icgc_api.get_manifest_id(fileids[0], api_url, repos)
+        except RuntimeError:
+            raise click.Abort
+    else:
+        try:
+            manifest_json = icgc_api.get_manifest(fileids, api_url, repos)
+        except RuntimeError:
+            raise click.Abort
+    if not manifest_json["unique"] or len(manifest_json["entries"]) != 1:
+        fi_ids = []
+        for repo_info in manifest_json["entries"]:
+            if repo_info["repo"] in REPOS:
+                for file_info in repo_info["files"]:
+                    if file_info["id"] in fi_ids:
+                        logger.error("Supplied manifest has repeated file identifiers.  Please specify a " +
+                                     "list of repositories to prioritize")
+                        raise click.Abort
+                    else:
+                        fi_ids.append(file_info["id"])
 
-    if repo == 'ega':
+    for repo_info in manifest_json["entries"]:
+        repo = repo_info["repo"]
```

Comment:
This method is way to deep. Please factor out some helper methods. For example `calculate_size`, etc.